hack/build-go.sh

```diff
@@ -128,13 +131,16 @@ for arg in $args; do
 
             # build virtctl for all architectures if requested
             if [ "${BIN_NAME}" = "virtctl" -a "${KUBEVIRT_RELEASE}" = "true" ]; then
-                for arch in amd64 arm64; do
+                for arch in amd64 arm64 s390x; do
                     for os in linux darwin windows; do
                         if [ "${os}" = "windows" ]; then
                             extension=".exe"
                         else
                             extension=""
                         fi
+                        if [ "${arch}" = "s390x" ] && [ "${os}" != "linux" ]; then
```

Comment:
Why do we need this check?

Reply:
The loop goes over all possible OS from `linux`, `darwin` and `windows`, however s390x only supports `linux` from the list, so we want to skip the other 2 passes on s390x here, as otherwise the script would fail trying to build virtctl for windows/darwin on s390x.

For more context here is the other discussion on the topic: https://github.com/kubevirt/kubevirt/pull/10490/files/4837a8fa7e08bb5a49375eabdec8afc5c2887454..b535187bec73c2348f1e50304b322655ed1e7853#r1370289684